cli/tests/environment-init.bats

```diff
@@ -177,6 +177,48 @@ function check_with_dir() {
   FLOX_SHELL=zsh "$FLOX_BIN" activate --trust -r "$OWNER/$NAME" -- python -c "import requests"
 }
 
+# ---------------------------------------------------------------------------- #
+
+# bats test_tags=init:go:module
+@test "'flox init' sets up a working Go module environment that works across all methods of activate" {
+  OWNER="owner"
+  NAME="name"
+  GO_BUILD_COMMAND="go build ."
+
+  cat > go.mod <<EOF
+  module go-module
+
+  go 1.21.0
+  EOF
+
+  cat > main.go <<EOF
+  package main
+
+  func main() {}
+  EOF
+
+  "$FLOX_BIN" init --auto-setup --name "$NAME"
+
+  FLOX_SHELL=bash "$FLOX_BIN" activate -- $GO_BUILD_COMMAND
+  FLOX_SHELL=zsh "$FLOX_BIN" activate -- $GO_BUILD_COMMAND
+
+  floxhub_setup "$OWNER"
+
+  "$FLOX_BIN" push --owner "$OWNER"
+
+  "$FLOX_BIN" delete -f
+
+  "$FLOX_BIN" pull "$OWNER/$NAME"
+
+  FLOX_SHELL=bash "$FLOX_BIN" activate -- $GO_BUILD_COMMAND
+  FLOX_SHELL=zsh "$FLOX_BIN" activate -- $GO_BUILD_COMMAND
+
+  "$FLOX_BIN" delete -f
+
+  FLOX_SHELL=bash "$FLOX_BIN" activate --trust -r "$OWNER/$NAME" -- $GO_BUILD_COMMAND
+  FLOX_SHELL=zsh "$FLOX_BIN" activate --trust -r "$OWNER/$NAME" -- $GO_BUILD_COMMAND
+}
```

Comment:
nonblocking (no need to address this before we merge): we only test each activate method for one language, and then all the other languages we test in `tests/$language.bats` generally against just one activation method. We don't run the tests in `tests/` against every commit pushed to a PR since they tend to be on the slow side, whereas the ones in `cli/tests` we do run on every commit

Reply:
Got it! I had already started with the `tests/go.bats` approach, but didn't push it since I wanted to see if this last commit made sense at all.